### PR TITLE
ENH / set up unsupported module versions more tightly

### PR DESCRIPTION
**Pull request by dnsl48**
related https://github.com/silverstripeltd/open-sourcerers/issues/65

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,24 +16,24 @@
         "silverstripe/bambusa-theme": "dev-master",
         "silverstripe/recipe-content-blocks": "^2.3",
         "silverstripe/crontask": "^2.1.2",
-        "silverstripe/dynamodb": "^4.0.1",
+        "silverstripe/dynamodb": "~4.0.1",
```
Comment (maxime-rainville) [inline composer.json]: "silverstripe/dynamodb" might has well be a supported module.

File: composer.json [hunk 2]
```diff
@@ -16,24 +16,24 @@
...
+        "sminnee/tagmanager": "~1.0.0",
         "sminnee/silverstripe-amplitude": "dev-master",
         "sminnee/silverstripe-asknicely": "^1.0@dev",
-        "benmanu/silverstripe-simple-styleguide": "^2.0",
+        "benmanu/silverstripe-simple-styleguide": "~2.0.5",
         "silverstripe/login-forms": "^4.0@beta",
-        "vulcandigital/silverstripe-seo": "^1.0",
-        "silverstripe/raygun": "^3.1",
+        "vulcandigital/silverstripe-seo": "~1.0.3",
+        "silverstripe/raygun": "~3.1.0",
```
Comment (maxime-rainville) [inline composer.json]: "silverstripe/raygun" and "sminnee/tagmanager" are not officially supported, but I think it's safe to assume that we can trust the maintainers to stick to semver.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
I would keep `silverstripe/dynamodb` with a loose constraint.

The other ones I'm not too fuss about.

File: composer.json (hunk 1 above)
Comment (dnsl48) [reply composer.json]: Ping @chillu 
I also reckon dynamodb must be a supported module, but it's not just yet :)

Comment (chillu) [reply composer.json]: That's irreleavnt to this pull request, right? I think a constraint to `~4.0` would be more appropriate. I think we can assume that module sticks to semver

File: composer.json (hunk 2 above)
Comment (chillu) [reply composer.json]: Yeah this is overkill, let's pull in minor releases automatically. There's no point getting notified via the new `composer outdated` if there's `raygun:3.2.0` available. I want to get notified when `raygun:4` is available. Minor releases should be pulled in as part of a `composer update` when it builds the image regularly.

File: composer.json (hunk 1 above)
Comment (maxime-rainville) [reply composer.json]: FYI `~4.0` is functionality equivalent to `^4.0.0`.

`~4.0.0` will let you upgrade to `4.0.1` but not `4.1.0`. `~4.0` and `^4.0.0` will both upgrade you to `4.1.0`

**Comment by dnsl48** [comment]
@maxime-rainville I've added the discussed changes, also added a fix for `bin/create-slack-message.php` so it only notifies about stable major upgrades

**Review by maxime-rainville — APPROVED** [review]

**Pull request merged**